manifests/prepare.pp

```diff
@@ -52,14 +54,103 @@
     }
   }
 
-  # TODO: manage server.cfg contents
-  if $::mcollective_configured {
-    file { $::agent_upgrade::params::mcodirs:
-      ensure => directory,
+  # manage client.cfg and server.cfg contents
+  file { $::agent_upgrade::params::mcodirs:
+    ensure => directory,
+  }
+  # The mco_*_config facts will return the location of mcollective config (or nil), prefering PE over FOSS.
+  if $::mco_server_config {
+    $mco_server = $::agent_upgrade::params::mco_server
+    file { $mco_server:
+      ensure  => file,
+      source  => $::mco_server_config,
+    }
+
+    if $::mco_server_settings {
+      $mco_server_libdir = $::mco_server_settings['libdir']
+      if $mco_server_libdir {
+        $mco_server_libdir_array = split($mco_server_libdir, $::agent_upgrade::params::path_separator)
+        # Only add the new path if it's not already in libdir
+        if [] == $mco_server_libdir_array.filter |$x| { $x == $::agent_upgrade::params::mco_libdir } {
+          ini_setting { 'server/libdir':
+            section => '',
+            setting => 'libdir',
+            path    => $mco_server,
+            value   => "${mco_server_libdir}:$::agent_upgrade::params::mco_libdir",
```

Comment:
This should use `$::agent_upgrade::params::path_separator` instead of hard-coded colon, but can wait until we add windows support.